src/components/TargetForm/TargetFormContent.js

```diff
@@ -69,65 +71,66 @@ const TargetFormContent = ({
     topicsError,
     loadingTopics,
     topicPlaceholder,
+    visible,
+    initialValues,
   });
 
+  const onEndEditingArea = () => {
+    setValues({
+      ...values,
+      areaLength: values.areaLength ? `${values.areaLength} m` : initialValues.areaLength,
+    });
+  };
+
   return (
     <>
       <Input
         label={existent ? areaEdit : area}
         testID="area-input"
         keyboardType="numeric"
-        error={errors[FIELDS.areaLength]}
-        invalid={targetError || errors[FIELDS.areaLength]}
         short={false}
         additionalStyles={styles.inputContainer}
-        placeholder={areaPlaceholder}
-        placeholderTextColor={BLACK}
         onBlur={areaInputProps.onBlur}
         onChangeText={areaInputProps.onChangeText}
-        value={existent ? existent.radius.toString() : areaInputProps.value}
+        value={existent ? `${existent.radius.toString()} m` : areaInputProps.value}
+        editable={!existent}
+        onFocus={() => setValues({ ...values, areaLength: '' })}
+        onEndEditing={onEndEditingArea}
       />
       <Input
         label={titleLabel}
         testID="target-title-input"
-        error={errors[FIELDS.title]}
-        invalid={targetError || errors[FIELDS.title]}
+        invalid={!existent && (networkError || errors[FIELDS.title])}
```

Comment:
How you get this networkError ? Is the one after submit? In that case, you can handle this as an `error` object like `useForm`  hook does

Reply:
Yes, I'm using the error object from the useForm as the initial value for the networkError, and setting it to `null` when the form is hidden. That's the usage of this networkError